agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -195,10 +195,16 @@ def container_outdated?(service_container)
     # @param [Docker::Container] service_container
     # @return [Boolean]
     def image_outdated?(service_container)
-      image = Docker::Image.get(service_pod.image_name) rescue nil
-      return true unless image
-      return true if image.id != service_container.info['Image']
-
+      begin
+        image = Docker::Image.get(service_pod.image_name)
+        if image.id != service_container.info['Image']
+          info "containers image(#{service_container.info['Image']} does not match images id(#{image.id})}"
+          return true
+        end
+      rescue Docker::Error::NotFoundError
+        info "failed to get image: #{service_pod.image_name}"
```

Comment:
I'm not saying that this `rescue` code here is broken... but it feels impossible for it to ever happen, so I'm not sure what the agent really should do if it does happen. Better to just let it fail, unless we know how this could ever happen and how the agent should deal with it in that scenario. This `return true` assumes that the `ServicePods::Creator` -> `Docker::Container.create` with that image might still succeed.

Using the `Docker::Image` returned by the `ImagePullWorker` would avoid these "impossible" errors.